### ISSUE TITLE
Allow for multiple stages for closed issues

### ISSUE THREAD

**Issue by ac21** [codetree/feedback#314]
## Background
Some users have a Github workflow that requires multiple stages for "closed" tickets.  For instance, when an issue is closed (via a pull request being committed), this could move to a "QA" stage

## Solution
Allow for the definition of custom stages for closed tickets

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/search/conversations/21235352299
+1

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/search/conversations/21307193208?q=Done
+1

**Comment by oskarszoon** [comment]
This is mostly because developers are used to using the Github `closes #xxx` workflow, but we still want an extra step/column once an issue is closed.

https://help.github.com/en/articles/closing-issues-using-keywords

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/2848986/conversations/21315764937
+1

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/2848986/conversations/21875410709
+1

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/2848986/conversations/21968642639
+1
however, there are some unique considerations:

> For merged PR's, you want to have the ability to place them in a "merged" stage, then finally into a "deployed" stage, correct?  Similarly, you want to be able to have issues flow into a "merged" stage, then into a "deployed" stage, and only once they hit deployed stage should they close, correct?

**Comment by CatalinaMoisuc** [comment]
> https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/2848986/conversations/21968642639
> +1
> however, there are some unique considerations:
> 
> > For merged PR's, you want to have the ability to place them in a "merged" stage, then finally into a "deployed" stage, correct?  Similarly, you want to be able to have issues flow into a "merged" stage, then into a "deployed" stage, and only once they hit deployed stage should they close, correct?

@ac21 what if we could just choose the label that we want the issue/PR to have when merged (similar to existent settings for Issue Rules->When an issue is closed) and match the label with a stage.

As for deployed stages,
https://help.github.com/en/articles/viewing-deployment-activity-for-your-repository
Deployment environments would be handy in this case since GitHub already knows which PR is deployed in each environment, and then if the PR is linked to an issue, you would already know which issues should be moved to which column based on the environment name.

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/2848986/conversations/23011254143
+1

**Comment by ac21** [comment]
+1 from @sfitts

**Comment by a-stankevich** [comment]
+1 from me:)

**Comment by ac21** [comment]
https://app.intercom.io/a/apps/he4q7g1x/inbox/inbox/2848986/conversations/24561958908
+ 1

**Comment by CatalinaMoisuc** [comment]
@ac21 any timeline or news on this feature?

**Comment by ac21** [comment]
@catalinamoisuc - this slipped this sprint, but hoping to finish it in may

**Comment by CatalinaMoisuc** [comment]
> @CatalinaMoisuc - this slipped this sprint, but hoping to finish it in may

@ac21 awesome news, really looking forward to this feature. 🙌🏻

**Comment by CatalinaMoisuc** [comment]
@ac21 any progress on this or was it postponed to end of June?

**Comment by CatalinaMoisuc** [comment]
@ac21 is there any hope that this feature will be implemented?
